### PR TITLE
Update how Vault Agent configures Consul Templates internal logger

### PR DESCRIPTION
**Pull request by catsby**
Builds off of https://github.com/hashicorp/vault/pull/7739 and fixes https://github.com/hashicorp/vault/issues/7804. Do not merge until #7739 is merged.

### CODE REVIEW

File: command/agent/template/template.go [hunk 1]
```diff
@@ -80,7 +92,12 @@ func (ts *Server) Run(ctx context.Context, incoming chan string, templates []*ct
 	// construct a consul template vault config based the agents vault
 	// configuration
 	var runnerConfig *ctconfig.Config
-	if runnerConfig = newRunnerConfig(ts.config, templates); runnerConfig == nil {
+	var runnerConfigErr error
+	if runnerConfig, runnerConfigErr = newRunnerConfig(ts.config, templates); runnerConfigErr != nil {
+		ts.logger.Error("template server failed to generate runner config", "error", runnerConfigErr)
+		return
+	}
+	if runnerConfig == nil {
```
Comment (calvn) [inline command/agent/template/template.go]: This check might be unneeded since it looks like it's not possible for `newRunnerConfig` to return `nil, nil`.

Comment (catsby) [reply command/agent/template/template.go]: Indeed, thanks! Removed in 6e3dbb9e2

File: command/agent/template/template.go [hunk 2]
```diff
@@ -25,6 +28,15 @@ type ServerConfig struct {
...
 
 	Namespace string
+
+	// LogLevel is needed to set the internal Consul Template Runner's log level
+	// to match the log level of Vault Agent. The internal Runner creates it's own
+	// logger and can't be set externally or copied from the Template Server.
+	//
+	// LogWriter is needed to initialize Consul Template's internal logger to use
+	// the same io.Writer that Vault Agent itself is using.
+	LogLevel  log.Level
```
Comment (kalafut) [inline command/agent/template/template.go]: Using `hclog` instead of `log` here and in `logLevelToStringPtr` will avoid a second import of `go-hclog`.

**Comment by kalafut** [comment]
Just the one import comment, otherwise LGTM.

File: command/agent/template/template.go (hunk 2 above)
Comment (catsby) [reply command/agent/template/template.go]: Great catch, thanks! I rebased and cleaned up the PR in 52272b7c4

**Pull request merged**